packages/components/src/Datalist/Datalist.component.js

```diff
@@ -145,9 +145,10 @@ function Datalist(props) {
 		}
 
 		if (entry.value !== value) {
-			setValue(entry.value);
+			setEntry(entry);
 		}
-		if (entry.name !== filterValue) {
+		// Update the input value only if user did not change it
+		if (!name || name === filterValue) {
```

Comment:
filterValue filters on `name`, not on `value`...

Reply:
yes, value is the technical id, name is the label (which you use in the filter input).. don't blame me, the new DS version of the component will be way better